.github/workflows/test.yml

```diff
@@ -3,13 +3,16 @@ name: BPX-tests
 on:
   workflow_dispatch:
   pull_request:
+  push:
+    branches:
+      - main
```

Comment:
CI should always run on the default branch. Things like codecov will not work if there is no coverage report uploaded from the default branch. See how [codecov says "no reports found"](https://app.codecov.io/gh/FaradayInstitution/BPX)